File: src/sort.py
from cstdio import *
from cstring import *
def main():
    array = [None] * 12
    array[0] = 14
    array[1] = 10
    array[2] = 18
    array[3] = 16
    array[4] = 20
    array[5] = 26
    array[6] = 23
    array[7] = 29
    array[8] = 26
    array[9] = 35
    array[10] = 32
    array[11] = 37
    insertSort(array, 0, 11)
    i = 0
    while i < 12:
        printf("%d ", array[i])
        i  =  i + 1
    return 0

def insertSort(data, l, r):
    auxiliary = 0
    i = l + 1
    while i <= r:
        if data[i] < data[i-1]:
            auxiliary = data[i]
            j = i-1
            while j >= l and data[j] > auxiliary:
                data[j + 1] = data[j]
                j -= 1
            data[j + 1] = auxiliary
        i = i + 1

if __name__ == '__main__':
    main()

File: src/test/sort.c
#include <stdio.h>

void insertSort(int data[], int l, int r) {
    int auxiliary = 0;
    for (int i = l+1; i <= r; i++) {
        if (data[i] < data[i - 1]) {
	    auxiliary = data[i];
            int j = i - 1;
            while (j >= l && data[j] > auxiliary) {
                data[j + 1] = data[j];
                j -= 1;
            }
            data[j + 1] = auxiliary;
        }
    }
}
int main(){
    int array[12] = { 14, 10, 18, 16, 20, 26, 23, 29, 26, 35, 32, 37 };
    insertSort(array, 0,11);
    for (int i = 0; i < 12; ++i) printf("%d ", array[i]);
    return 0;
}
